components/nodemanager-service/pgdb/node_managers_cloud_integration_test.go

```diff
@@ -122,6 +122,32 @@ func (suite *NodeManagersAndNodesDBSuite) TestUpdateOrInsertInstanceSourceStateI
 	suite.Equal(0, len(terminatedNodes))
 }
 
+func (suite *NodeManagersAndNodesDBSuite) TestUpdateOrInsertInstanceSourceStateInDbIngestingNewNodeIncludesNameAndMakesMgrConnection() {
+	mgr := manager.NodeManager{Name: "test", Type: "azure-vm"}
+	mgrID, err := suite.Database.AddNodeManager(&mgr, "12345678")
+	if err != nil {
+		suite.FailNow(err.Error())
+	}
+	instanceState := pgdb.InstanceState{ID: "i-079356", Name: "my-new-node", State: "running", Region: "eu-west-1"}
+	_, err = suite.Database.UpdateOrInsertInstanceSourceStateInDb(instanceState, mgrID, "12345678", "azure-vm")
+	if err != nil {
+		suite.FailNow(err.Error())
+	}
+	filter := &common.Filter{
+		Key:    "manager_id",
+		Values: []string{mgrID},
+	}
+	mgrNodes, _, err := suite.Database.GetNodes("", nodes.Query_ASC, 1, 100, []*common.Filter{filter})
+	if err != nil {
+		suite.FailNow(err.Error())
+	}
+	suite.Equal(1, len(mgrNodes))
+
+	node := mgrNodes[0]
```

Comment:
If the assertion above fails (ie, if `len(mgrNodes) == 0`) this line will panic with a null pointer exception. If line 144 is changed to use `suite.Require().Equal(...)` the test will stop on that line instead.